checker/client.go

```diff
@@ -49,14 +54,28 @@ func GetClients(ctx context.Context, repoURI, localURI string, logger *log.Logge
 			retErr
 	}
 
-	githubRepo, errGitHub := ghrepo.MakeGithubRepo(repoURI)
-	if errGitHub != nil {
-		return githubRepo,
-			nil,
-			nil,
-			nil,
-			nil,
-			fmt.Errorf("getting local directory client: %w", errGitHub)
+	_, experimental := os.LookupEnv("SCORECARD_EXPERIMENTAL")
+
+	if strings.Contains(repoURI, "gitlab.") && experimental {
```

Comment:
maybe create a different function `makeRepo()`

Reply:
A GitLab self-hosted deployment could be named anything, not sure if this check is valid. If repoURI had to match exactly `gitlab.com`, we'd only support GitLab.com. 

Another approach that doesn't rely on domain names could be a `--gitlab-repo` flag.

Reply:
> A GitLab self-hosted deployment could be named anything, not sure if this check is valid. If repoURI had to match exactly `gitlab.com`, we'd only support GitLab.com.
> 
> Another approach that doesn't rely on domain names could be a `--gitlab-repo` flag.

What about checking if the repository host responds to the gitlab api? For example `https://gitlab.com/api/v4/version` that endpoint does require authentication, but there are some that don't.

Reply:
We can now detect self-hosted GitLab repos by checking if they respond to an unauthenticated API, also added tests to `clients/gitlabrepo/repo_test.go`.